### ISSUE TITLE
systemd-network fails to setup an interface with a static ip-address in a systemd-nspawn container

### ISSUE THREAD

**Issue by ostcar** [systemd/systemd#396]
I tried to set up an interface inside a systemd-nspawn container (started with --network-veth) via systemd-network.

This works fine when the interface is configured with DHCP but fails when a static ip-address is used. The error messages are:

```
systemd-networkd[128]: host0: Cannot configure IPv6 forwarding for interface: Read-only file system
systemd-networkd[128]: host0: Cannot configure IPv4 forwarding for interface host0: Read-only file system
```

/proc/sys is mounted read only (as described in man systemd-nspawn).

I used the following network configuratoin

```
[Match]
Name=host0

[Address]
Address=192.168.10.2/24
```

I also tried to set `IPForward` to `no` or `kernel` which made no difference.

The host and the container system are running with systemd 221

**Comment by ghost** [comment]
Can confirm this issue. Is there any fix or workaround available?

**Comment by ghost** [comment]
It seems to work with version _222_. Steps to "reproduce" it:

# Dependencies

I installed `systemd` in version _222_, using the _testing_ repository in _Arch Linux_. 

# Configuration

## Host-configuration

My host-configuration looks like

`10-intel.network`:

``` bash
[Match]
Name=enp3s0

[Network]
Bridge=br0
```

`50-br0.netdev`:

``` bash
[NetDev]
Name=br0
Kind=bridge
```

`50-br0.network`:

``` bash
[Match]
Name=br0

[Network]
DNS=192.168.1.1
Address=192.168.1.111/25
Gateway=192.168.1.1
```

## Container-configuration

Inside the container I enabled `systemd-networkd` (no other stuff like `dhcpcd`). The configuration for `host0` looks like:

``` bash
[Match]
Name=host0

[Network]
DNS=192.168.1.1
Gateway=192.168.1.1
Address=192.168.1.112/25
```

# Tests

Now we can use `networkctl status -a` to show the network status.

## Host

The output will be:

``` bash
● 3: enp3s0
       Link File: /usr/lib/systemd/network/99-default.link
    Network File: /etc/systemd/network/10-intel.network
            Type: ether
           State: routable (configured)
            Path: pci-0000:03:00.0
          Driver: e1000e
          Vendor: Intel Corporation
           Model: 82574L Gigabit Network Connection (Gigabit CT Desktop Adapter)
      HW Address: 68:05:ca:23:16:14 (Intel Corporate)
             MTU: 1500
         Address: 192.168.1.101
                  fe80::6a05:caff:fe23:1614
         Gateway: 192.168.1.1
             DNS: 192.168.1.1

● 4: br0
       Link File: /usr/lib/systemd/network/99-default.link
    Network File: /etc/systemd/network/50-br0.network
            Type: ether
           State: routable (configured)
          Driver: bridge
      HW Address: ae:2d:49:84:77:99
             MTU: 1500
         Address: 192.168.1.111
                  fe80::ac2d:49ff:fe84:7799
         Gateway: 192.168.1.1 (Arcadyan Technology Corporation)
             DNS: 192.168.1.1

● 5: vb-stefan-test
       Link File: /usr/lib/systemd/network/99-default.link
    Network File: n/a
            Type: ether
           State: n/a (n/a)
          Driver: veth
      HW Address: 2a:30:0c:5e:29:74
             MTU: 1500
         Address: fe80::2830:cff:fe5e:2974
```

## Container

The `networkctl status -a` in the container looks like:

``` bash
* 2: host0
       Link File: n/a
    Network File: /etc/systemd/network/host0.network
            Type: ether
           State: routable (configured)
      HW Address: 56:d3:9f:df:1e:72
             MTU: 1500
         Address: 192.168.1.112
                  fe80::54d3:9fff:fedf:1e72
         Gateway: 192.168.1.1 (Arcadyan Technology Corporation)
             DNS: 192.168.1.1
```

# Symlinks

I followed @poettering [hint](http://lists.freedesktop.org/archives/systemd-devel/2015-April/031395.html) for creating symlinks on both the host and the container:

``` bash
ln -sf /dev/null /etc/systemd/network/80-container-host0.network
```

This is also mentioned in the [Arch Linux](https://wiki.archlinux.org/index.php/Systemd-networkd#Usage_with_containers) wiki.

# Summary

With the tested `systemd` version _222_ the container network (with static configured) ip addresses should work :+1: 

@ostcar Could you please test it with a higher `systemd` version? With _222_ all the "Read-only file system" messages disappeared :)

I'm going to test that setup with ipv6 very soon :angel:

**Comment by poettering** [comment]
Sounds like the same issue as #469, which has been fixed with 15dee3f07c646fd345b0aa30c6566071b3365db7. Closing this one hence.
